### PR TITLE
Tools: Add tests for processing components

### PR DESCRIPTION
**Pull request by singalsu**
This patch adds Octave scripts for objective quality parameters checks for processing components. The standard tests include chirp spectrogram, gain, dynamic range, THD+N over frequency, and frequency response. 

There's still minor general issues/limitations (currently only test for flat response FIR/IIR, failure when testing volume), and need for more test topologies (will use @fredoh9  topology work). But since I feel this is generally useful for processing developers I'd like to share this early.

### CODE REVIEW

File: tools/test/audio/process_test.m [hunk 1]
```diff
@@ -0,0 +1,408 @@
...
+% Copyright(c) 2017-202 Intel Corporation. All rights reserved.
+% Author: Seppo Ingalsuo <seppo.ingalsuo@linux.intel.com>
+
+%% Defaults for call parameters
+if nargin < 1
+	comp = 'EQIIR';
+end
+
+if nargin < 2
+	bits_in_list = [16 32];
```
Comment (singalsu) [inline tools/test/audio/process_test.m:17]: 24 missing...

**Comment by singalsu** [comment]
@sebcarlucci I could not add you as reviewer. Maybe @lgirdwood can fix it.

**Comment by plbossart** [comment]
> @sebcarlucci I could not add you as reviewer. Maybe @lgirdwood can fix it.

reviewers need to be member of the project, invite sent to @sebcarlucci

**Comment by singalsu** [comment]
In current state the script outputs tables like this plus some plots those likely need to be by default suppressed to be scripted e.g. with Jenkins:

```
EQIIR test result: Gain (dB)
in \ out,       16,       24,       32
      16,    -0.00,        x,        x
      24,        x,    -0.00,        x
      32,        x,        x,    -0.00


EQIIR test result: Worst-case THD+N vs. frequency
in \ out,       16,       24,       32
      16,   -62.71,        x,        x
      24,        x,  -110.73,        x
      32,        x,        x,  -158.50


EQIIR test result: Dynamic range (dB CCIR-RMS)
in \ out,       16,       24,       32
      16,    83.34,        x,        x
      24,        x,   131.47,        x
      32,        x,        x,   178.78


EQIIR test result: Fails
in \ out,             16,             24,             32
      16,      0/0/0/0/0,              x,              x
      24,              x,      0/0/0/0/0,              x
      32,              x,              x,      0/0/0/0/0


Number of passed tests = 15
Number of failed tests = 0
Number of non-applicable tests = 0
Number of skipped tests = 0

Test passed.
res = 0

```

For the ASCII/CSV output I need to figure out layout for all tested channels and multiple sweeps like for THD+N. Now The script outputs worst-case performance found, except for gain where the table shows only ch1. The pass/fail checks are done for every channel. The criteria is also only "something" that needs more thinking.

**Comment by singalsu** [comment]
@lgirdwood Yep, need to have description and instructions in SOF docs.

@sebcarlucci All processing components by their designed characteristics can't pass all the tests (flat response 20 - 20 kHz => flat THD+N, non flat response => wost-case THD+N can look poor at frequencies where it doesn't matter) I need to add component specific relaxations to test criteria. Since this is not yet merged I can add some exceptions handling for DCB and EQs when run with non-flat response.

**Comment by lgirdwood** [comment]
CI timeout

**Pull request merged**